### ISSUE TITLE
Split up and test transformations

### ISSUE THREAD

**Issue by rerpha** [ess-dmsc/nexus-constructor#176]
Most of this is already done on the `108_Use_in-memory_NeXus_for_part_of_model` branch so just needs to be copied over and check that the tests pass. That branch will diverge a lot so I don't think we should wait on it being merged as it may eventually just be closed anyway. 

This is a step towards doing #175 as a lot of the tests that open files are creating and "testing" the transformations
